qiskit/algorithms/evolvers/pvqd/pvqd.py

```diff
@@ -0,0 +1,414 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The projected Variational Quantum Dynamics Algorithm."""
+
+from typing import Optional, Union, List, Tuple, Callable
+
+import logging
+import numpy as np
+
+from qiskit import QiskitError
+from qiskit.algorithms.optimizers import Optimizer, Minimizer
+from qiskit.circuit import QuantumCircuit, ParameterVector
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.extensions import HamiltonianGate
+from qiskit.providers import Backend
+from qiskit.opflow import OperatorBase, CircuitSampler, ExpectationBase, StateFn, MatrixOp
+from qiskit.synthesis import EvolutionSynthesis, LieTrotter
+from qiskit.utils import QuantumInstance
+
+from .pvqd_result import PVQDResult
+from .utils import _get_observable_evaluator, _is_gradient_supported
+
+from ..evolution_problem import EvolutionProblem
+from ..evolution_result import EvolutionResult
+from ..real_evolver import RealEvolver
+
+logger = logging.getLogger(__name__)
+
+
+class PVQD(RealEvolver):
+    """The projected Variational Quantum Dynamics (p-VQD) Algorithm.
+
+    In each timestep, this algorithm computes the next state with a Trotter formula
+    (specified by the ``evolution`` argument) and projects the timestep onto a variational form
+    (``ansatz``). The projection is determined by maximizing the fidelity of the Trotter-evolved
+    state and the ansatz, using a classical optimization routine. See Ref. [1] for details.
+
+    The following attributes can be set via the initializer but can also be changed once the
+    PVQD object has been constructed.
```

Comment:
> but can also be changed
Does that imply they can be read too - could infer it was just setters? Maybe `but can also be read/updated once...` or something like that